tests/kernel_bundle/sycl_join.h

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tools for sycl::join tests
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_SYCL_JOIN_H
+#define __SYCLCTS_TESTS_SYCL_JOIN_H
+
+#include "../common/common.h"
+
+namespace sycl_cts::tests::sycl_join {
+
+/** @brief Provides full comparison of device images in two kernel bundles
+ *  @tparam State sycl::bundle_state
+ *  @details Checks that rhs_kb contains all device images form lhs_kb.
+ */
+template <sycl::bundle_state State>
+inline bool check_dev_images_equal(sycl_cts::util::logger &log,
+                                   sycl::kernel_bundle<State> &lhs_kb,
+                                   sycl::kernel_bundle<State> &rhs_kb) {
+  bool ok = (std::distance(lhs_kb.begin(), lhs_kb.end()) ==
+             std::distance(rhs_kb.begin(), rhs_kb.end()));
+  if (!ok) return ok;
+  for (const auto &dev_img : lhs_kb) {
+    ok &= (std::find(rhs_kb.begin(), rhs_kb.end(), dev_img) != rhs_kb.end());
+  }
```

Comment:
This is another case where we would need https://gitlab.khronos.org/sycl/Specification/-/issues/586 to avoid this O(n^2) algorithm.

Reply:
OK, from the SYCL workgroup meeting today this might be fixed in the specification.
In the meantime you can keep the O(n^2) algorithm and we can come back on this.
The problem is to remember where we will have to fix it. Adding some kind of TODO or adding a link to each test in https://gitlab.khronos.org/sycl/Specification/-/issues/586 ?

Reply:
@keryell  I am going to leave the same comment as [here](https://github.com/KhronosGroup/SYCL-CTS/commit/a3afd778782dff127cabb8f71fe4d4036f3ee5c5#diff-d7dd04b2c33a3f23629a9f87a1e1089058ce7e6a7fa093db890c2d2508b4d206R141-R146)

If this comment structure is ok, then I'll apply it to this case

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/229/commits/05b37d37e6f10c1afd5ea287aaa95be0d52d9259#diff-6b84827637b50d648c97ff5af6c69a319ec4de83ebccd0b2ebe5b18d2774dbf6R27-R32)

Reply:
Good. Can you also retrofit from https://github.com/KhronosGroup/SYCL-CTS/pull/226/commits/d910c659140ff7236d0fdc6281a733d02b2b9a3c ?

Reply:
@keryell Ronan, if I correctly understood you then, I have already done it.
Updated comment can be found in [tests/kernel_bundle/sycl_join.h](https://github.com/KhronosGroup/SYCL-CTS/pull/229/files#diff-6b84827637b50d648c97ff5af6c69a319ec4de83ebccd0b2ebe5b18d2774dbf6R27-R32)